packages/engine/Source/Renderer/Context.js

```diff
@@ -1358,7 +1365,11 @@ function continueDraw(context, drawCommand, shaderProgram, uniformMap) {
       );
     }
   } else {
-    count = defaultValue(count, va.numberOfVertices);
+    if (count) {
+      count = count < va.numberOfIndices ? count : va.numberOfIndices;
```

Comment:
I think we should be comparing to `va.numberOfVertices` in this case.

Reply:
Fixed.